src/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -241,6 +241,74 @@ public static void ZipWithLargeSparseFile()
+            0x2E, 0x62, 0x69, 0x6E, 0x03, 0x00, 0x50, 0x4B, 0x01, 0x02, 0x14, 0x00, 0x14, 0x00, 0x06, 0x00,
+            0x08, 0x00, 0x00, 0x00, 0x21, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x16, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x78, 0x6C, 0x2F, 0x63, 0x75, 0x73, 0x74, 0x6F, 0x6D, 0x50, 0x72, 0x6F,
+            0x70, 0x65, 0x72, 0x74, 0x79, 0x32, 0x2E, 0x62, 0x69, 0x6E, 0x50, 0x4B, 0x05, 0x06, 0x00, 0x00,
+            0x00, 0x00, 0x01, 0x00, 0x01, 0x00, 0x44, 0x00, 0x00, 0x00, 0x36, 0x00, 0x00, 0x00, 0x00, 0x00
+        };
+        internal static readonly byte[] EmptyFileCompressedWrongSize =
+        {
+            0x50, 0x4B, 0x03, 0x04, 0x14, 0x00, 0x06, 0x00, 0x08, 0x00, 0x00, 0x00, 0x21, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x04, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x16, 0x00, 0x00, 0x00, 0x78, 0x6C,
+            0x2F, 0x63, 0x75, 0x73, 0x74, 0x6F, 0x6D, 0x50, 0x72, 0x6F, 0x70, 0x65, 0x72, 0x74, 0x79, 0x32,
+            0x2E, 0x62, 0x69, 0x6E, 0xBA, 0xAD, 0x03, 0x00, 0x50, 0x4B, 0x01, 0x02, 0x14, 0x00, 0x14, 0x00,
+            0x06, 0x00, 0x08, 0x00, 0x00, 0x00, 0x21, 0x00, 0x00, 0x00, 0x00, 0x00, 0x04, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x16, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x78, 0x6C, 0x2F, 0x63, 0x75, 0x73, 0x74, 0x6F, 0x6D, 0x50,
+            0x72, 0x6F, 0x70, 0x65, 0x72, 0x74, 0x79, 0x32, 0x2E, 0x62, 0x69, 0x6E, 0x50, 0x4B, 0x05, 0x06,
+            0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x01, 0x00, 0x44, 0x00, 0x00, 0x00, 0x38, 0x00, 0x00, 0x00,
+            0x00, 0x00
+        };
+        internal static readonly Collections.Generic.List<byte[]> issue30317 = new Collections.Generic.List<byte[]>(){
+            EmptyFileCompressedWithETX, EmptyFileCompressedWrongSize
+        };
+
+        /// <summary>
+        /// This test checks behavior of ZipArchive with unexpected zip files:
+        /// 1. EmptyFileCompressedWithEOT has 
+        /// Deflate 0x08, _uncompressedSize 0, _compressedSize 2, compressed data: 0x0300 (\u0003 ETX)
+        /// 2. EmptyFileCompressedWrongSize has 
+        /// Deflate 0x08, _uncompressedSize 0, _compressedSize 4, compressed data: 0xBAAD0300 (just bad data)
+        /// ZipArchive is expected to change compression method to Stored (0x00) and ignore "bad" compressed size
+        /// </summary>
+        [Fact]
+        public static void ReadArchive_WithEmptyDeflatedFile()
+        {
+            foreach (var fileBytes in issue30317)
+            {
+                using (var testStream = new MemoryStream(fileBytes))
+                {
+                    var tmpfilename = "xl/customProperty2.bin";
+                    var tmppathname = Path.GetTempPath();
+                    var tmpfilepath = tmppathname + tmpfilename;
```

Comment:
This should use `string tmpfilepath = GetTestFilePath();`, using the helper from the base test class.  That will ensure it's in a good temporary location with a good name, and that the file will be cleaned up after the test ends.  You can use `TestDirectory` to know what directory it's in.

Reply:
Oh, I see, you're not generating the file name, but rather it's already encoded into the original zip.  In that case, just use `TestDirectory` instead of `Path.GetTempPath()`.